L1Trigger/L1THGCalUtilities/python/caloTruthCells_cff.py

```diff
@@ -1,70 +1,89 @@
 import FWCore.ParameterSet.Config as cms
 
 from L1Trigger.L1THGCalUtilities.caloTruthCellsProducer_cfi import *
+from L1Trigger.L1THGCalUtilities.hgcalTriggerNtuples_cfi import *
 
-from L1Trigger.L1THGCal.hgcalTriggerGeometryESProducer_cfi import *
-from L1Trigger.L1THGCal.hgcalConcentrator_cff import *
-from L1Trigger.L1THGCal.hgcalBackEndLayer1_cff import *
-from L1Trigger.L1THGCal.hgcalBackEndLayer2_cff import *
-from L1Trigger.L1THGCal.hgcalTowerMap_cff import *
-from L1Trigger.L1THGCal.hgcalTower_cff import *
-
-hgcalTruthConcentratorProducer = hgcalConcentratorProducer.clone(
-    InputTriggerCells = cms.InputTag('caloTruthCellsProducer')
-)
-
-hgcalTruthBackEndLayer1Producer = hgcalBackEndLayer1Producer.clone(
-    InputTriggerCells = cms.InputTag('hgcalTruthConcentratorProducer:HGCalConcentratorProcessorSelection')
-)
-
-hgcalTruthBackEndLayer2Producer = hgcalBackEndLayer2Producer.clone(
-    InputCluster = cms.InputTag('hgcalTruthBackEndLayer1Producer:HGCalBackendLayer1Processor2DClustering')
-)
+caloTruthCells = cms.Sequence(caloTruthCellsProducer)
 
-hgcalTruthTowerMapProducer = hgcalTowerMapProducer.clone(
-    InputTriggerCells = cms.InputTag('caloTruthCellsProducer')
-)
+# Add the "full truth" multiclusters (each cluster made with all cells belonging to one gen particle) to the ntuples
 
-hgcalTruthTowerProducer = hgcalTowerProducer.clone(
-    InputTowerMaps = cms.InputTag('hgcalTruthTowerMapProducer:HGCalTowerMapProcessor')
+ntuple_multiclusters_fulltruth = ntuple_multiclusters.clone(
+    Multiclusters = cms.InputTag('caloTruthCellsProducer:HGCalBackendLayer2Processor3DClustering'),
+    Prefix = cms.untracked.string('cl3dfulltruth')
 )
+hgcalTriggerNtuplizer.Ntuples.append(ntuple_multiclusters_fulltruth)
 
-caloTruthCells = cms.Sequence(
-    caloTruthCellsProducer *
-    hgcalTruthConcentratorProducer *
-    hgcalTruthBackEndLayer1Producer *
-    hgcalTruthBackEndLayer2Producer *
-    hgcalTruthTowerMapProducer *
-    hgcalTruthTowerProducer
-)
+# If caloTruthCellsProducer.makeCellsCollection is True, can run the clustering algorithm over output cells too
 
-from L1Trigger.L1THGCalUtilities.hgcalTriggerNtuples_cfi import *
+if caloTruthCellsProducer.makeCellsCollection:
+    ## cluster and tower sequence
 
-ntuple_triggercells.caloParticlesToCells = cms.InputTag('caloTruthCellsProducer')
-ntuple_triggercells.FillTruthMap = cms.bool(True)
+    from L1Trigger.L1THGCal.hgcalTriggerGeometryESProducer_cfi import *
+    from L1Trigger.L1THGCal.hgcalConcentrator_cff import *
```

Comment:
Here and in the following lines I would rather import from  `*Producer_cfi` files so that you don't redefine the sequences. This would otherwise break the "chains" definition done with the `hgcalTriggerChains` tool.

Reply:
Good point. Fixed.